typescript/agentkit/src/wallet-providers/cdpV2EvmWalletProvider.ts

```diff
@@ -0,0 +1,272 @@
+import { CdpClient, EvmServerAccount } from "@coinbase/cdp-sdk";
+import {
+  Abi,
+  Address,
+  ContractFunctionArgs,
+  ContractFunctionName,
+  createPublicClient,
+  Hex,
+  http,
+  PublicClient,
+  ReadContractParameters,
+  ReadContractReturnType,
+  serializeTransaction,
+  TransactionRequest,
+  TransactionSerializable,
+} from "viem";
+import { Network, NETWORK_ID_TO_CHAIN_ID, NETWORK_ID_TO_VIEM_CHAIN } from "../network";
+import { EvmWalletProvider } from "./evmWalletProvider";
+import { WalletProviderWithClient, CdpV2WalletProviderConfig } from "./cdpV2Shared";
+
+interface ConfigureCdpV2EvmWalletProviderWithWalletOptions {
+  /**
+   * The CDP client of the wallet.
+   */
+  cdpClient: CdpClient;
```

Comment:
```suggestion
  cdp: CdpClient;
```

Reply:
We like naming the client variable just `cdp`, would be nice to follow that in AgentKit!